src/compiler/checker.ts

```diff
@@ -2465,30 +2471,62 @@ module ts {
             }
         }
 
-        // Return combined list of type parameters from all declarations of a class or interface. Elsewhere we check they're all
-        // the same, but even if they're not we still need the complete list to ensure instantiations supply type arguments
-        // for all type parameters.
-        function getTypeParametersOfClassOrInterface(symbol: Symbol): TypeParameter[] {
+        function appendTypeParameters(typeParameters: TypeParameter[], declarations: TypeParameterDeclaration[]): TypeParameter[] {
+            for (let declaration of declarations) {
+                let tp = getDeclaredTypeOfTypeParameter(getSymbolOfNode(declaration));
+                if (!typeParameters) {

```

Comment:
Based on what this function does, it looks like it should really be written as:

```ts
function appendTypeParameters(declarations: TypeParameterDeclaration[], typeParameters?: TypeParameters[]) {
...
```

The declarations are the actual input.  The 'typeParameters' array is the output.  We don't generally put output parameters before intput parameters.  Also, as the second parameter is optional (based on the ```!typeParameters``` check), we should likely mark it as such.